### PR TITLE
chore: regex free isAbsoluteURL

### LINKED ISSUE #4037: Alternative isAbsoluteURL solution
https://github.com/webpack/webpack-dev-server/blob/497e615f24d5a16266f0faa8db1063f1b0ec2b82/lib/Server.js#L53-L62
```js
 static isAbsoluteURL(url) { 
   try { return new URL(url), true } catch(e) {}
   return false
 } 
```
then you can use both a URL class and a strings and weird domain names that are very special

### PR DESCRIPTION
**Pull request by jimmywarting**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

using existing tests

### Motivation / Use-Case

Domain names can be tricky with weird names... like punycode

### Breaking Changes

It parse url a bit differently now. It can throw when the url is invalid...

### Additional Info

closes #4037

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -50,15 +50,13 @@ class Server {
     };
   }
 
-  static isAbsoluteURL(URL) {
-    // Don't match Windows paths `c:\`
-    if (/^[a-zA-Z]:\\/.test(URL)) {
+  static isAbsoluteURL(uri) {
+    try {
+      new URL(uri); // eslint-disable-line
```
Comment (snitin315) [inline lib/Server.js:55]: Let's only disable the specific rule.

**Comment by snitin315** [comment]
@jimmywarting can you fix CI?

**Comment by jimmywarting** [comment]
I'm not really sure what needs to be fixed...

**Comment by snitin315** [comment]
The following two cases are failing.

https://github.com/webpack/webpack-dev-server/blob/a27b55389108a1b212fab5a39fe01b083fb27c94/test/e2e/static-directory.test.js#L521-L526

https://github.com/webpack/webpack-dev-server/blob/a27b55389108a1b212fab5a39fe01b083fb27c94/test/e2e/static-directory.test.js#L535-L540

**Comment by alexander-akait** [comment]
Yes, I see, looks like we need check on `isWindows`

**Comment by alexander-akait** [comment]
We decide do not change logic, because no benefit, please read https://github.com/webpack/webpack-dev-server/pull/4074, anyway thanks for PR

**Pull request closed without merging**